test/functional/unified-spec-runner/match.ts

```diff
@@ -235,32 +248,73 @@ export function specialCheck(
   }
 }
 
+// CMAP events where the payload does not matter.
+const EMPTY_CMAP_EVENTS = {
+  poolCreatedEvent: ConnectionPoolCreatedEvent,
+  poolClosedEvent: ConnectionPoolClosedEvent,
+  connectionCreatedEvent: ConnectionCreatedEvent,
+  connectionReadyEvent: ConnectionReadyEvent,
+  connectionCheckOutStartedEvent: ConnectionCheckOutStartedEvent,
+  connectionCheckOutFailedEvent: ConnectionCheckOutFailedEvent,
+  connectionCheckedOutEvent: ConnectionCheckedOutEvent,
+  connectionCheckedInEvent: ConnectionCheckedInEvent
+};
+
+function validEmptyCmapEvent(
+  expected: ExpectedCommandEvent | ExpectedCmapEvent,
+  actual: CommandEvent | CmapEvent
+) {
+  return Object.keys(EMPTY_CMAP_EVENTS).some(key => {
+    const eventType = EMPTY_CMAP_EVENTS[key];
+    return actual instanceof eventType;
+  });
+}
+
 export function matchesEvents(
-  expected: ExpectedEvent[],
-  actual: CommandEvent[],
+  expected: (ExpectedCommandEvent & ExpectedCmapEvent)[],
+  actual: (CommandEvent & CmapEvent)[],
   entities: EntitiesMap
 ): void {
   // TODO: NodeJS Driver has extra events
   // expect(actual).to.have.lengthOf(expected.length);
+  if (actual.length !== expected.length) {
+    //const actualNames = actual.map(a => a.constructor.name);
+    const expectedNames = expected.map(e => Object.keys(e)[0]);
+    console.log(
+      `Expected event count mismatch, expected ${inspect(expectedNames)} but got ${inspect(actual)}`
```

Comment:
Is this supposed to be an error now or is this just helping debug?

Reply:
This is just helping debug at the moment.

Reply:
oh sorry this should be a `expect.fail`

Reply:
Can we assert the lengths match in the commented out assertion above and we can still use this string as the message for that assertion?

Reply:
Updated for that commented out assertion.

Reply:
Was it intentional to leave these commented out? I thought @mbroadst identified that we can assert the length check now?

Reply:
It was intentional as the Node driver does not execute a `killCursors` after a `getMore` has errored, which causes the absence of the `killCursors` events in some cases. I could not find anywhere so far in the specs that states an errored `getMore` requires an immediate `killCursors` afterwards, and all the driver does in this case is cleanup the cursor (but not explicitly `close()` it).